newIDE/app/src/GameDashboard/GameDetailsDialog.js

```diff
@@ -165,11 +172,40 @@ export const GameDetailsDialog = ({
         orientation: project.getOrientation(),
         // The thumbnailUrl is updated only when a build is made public.
       });
-      const authorAcls = getAclsFromAuthorIds(project.getAuthorIds());
-      await setGameUserAcls(getAuthorizationHeader, id, gameId, authorAcls);
+      try {
+        const authorAcls = getAclsFromUserIds(
+          project.getAuthorIds().toJSArray()
+        );
+        const ownerAcls = getAclsFromUserIds(partialGameChange.ownerIds);
+        await setGameUserAcls(getAuthorizationHeader, id, gameId, {
+          ownership: ownerAcls,
+          author: authorAcls,
+        });
+      } catch (error) {
+        console.error('Unable to update the game owners or authors:', error);
+        showErrorBox({
+          message:
+            i18n._(t`Unable to update the game owners or authors.`) +
+            ' ' +
+            i18n._(t`Verify your internet connection or try again later.`),
+          rawError: error,
+          errorId: 'game-acls-update-error',
+        });
+      }
+      // Set public game to null to have it been refetched by the callback above.
+      // TODO this probably needs a refactor.
```

Comment:
Can you explain what/why is this refactoring needed, what is the suboptimal behavior that results from this?

Reply:
`publicGame` is set to null knowing that another component will detect this as an initialization state and fetch the Game again.
This component should either ask for game fetching or do it itself and signal the update.

Reply:
actually the `publicGame` being set to null is here to trigger the show of a loader by the top component (because there is no public game, it considers it is loading)
It is the `onGameUpdated` that triggers the refetch. so, not that bad?

Reply:
ok